File: src/Net/1. First Course/1. Initial Tee/RefactoringGolf.Store/Manager.cs
namespace RefactoringGolf.Store
{
    public class Manager : Employee
    {
        public Manager(string firstName, string lastName, decimal fixedSalary)
            : base(firstName, lastName, fixedSalary)
        {
        }
        
        public decimal SalaryAfterAdditionsAndDeductions()
        {
            decimal benefits = SalaryBenefits();
            decimal pensionFounds = this.FixedSalary * 10 / 100;
            decimal tax = 0;
            if (FixedSalary > 3500)
                tax = FixedSalary * 5 / 100;
            return this.FixedSalary + benefits - pensionFounds - tax;
        }

        private decimal SalaryBenefits()
        {
            return this.subordinates.Count * 20;
        }
    }
}

File: src/Java/1. First Course/3. Third Hole/src/main/refactoringgolf/store/Manager.java
package refactoringgolf.store;

public class Manager extends Employee {

	public Manager(String firstName, String lastName, float fixedSalary) {
		super(firstName, lastName, fixedSalary);
	}

	public float salaryAfterAdditionsAndDeductions() {
		float benefits = salaryBenefits();
		float pensionFounds = this.fixedSalary * 10 / 100;
		float tax = 0;
		if (fixedSalary > 3500)
			tax = fixedSalary * 5 / 100;
		return fixedSalary + benefits - pensionFounds - tax;
	}

	private float salaryBenefits() {
		return this.subordinates.size() * 20;
	}
}
